src/lib/transit/planning/path-finder.ts

```
@@ -0,0 +1,242 @@
+/**
+ * Additional utility for weighted A* search - for transit networks where
+ * transfer penalties or wait times might be important
+ */
+export class WeightedAStarPathFinder<T> extends AStarPathFinder<T> {
+  constructor(
+    getKey: (node: T) => string,
+    getNeighbors: (node: T) => T[],
+    heuristic: (node: T, goal: T) => number,
+    getCost: (from: T, to: T) => number,
+    isGoal: (node: T, goal: T) => boolean,
+    private heuristicWeight: number = 1.2 // Weight > 1 makes algorithm favor goals
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)